modules/core/src/v2/keycard.ts

```diff
@@ -0,0 +1,381 @@
+/**
```

Comment:
Is this an entirely new file, or was it copied from somewhere else?

Reply:
It's a new file, but the code was extracted out of wallet.ts